indico/modules/events/papers/controllers/common.py

```diff
@@ -0,0 +1,72 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+import os
+
+from flask import flash, request, session
+
+from indico.modules.events.papers.forms import BulkPaperJudgmentForm
+from indico.modules.events.papers.models.revisions import PaperRevisionState
+from indico.modules.events.papers.operations import judge_paper
+from indico.modules.events.util import ZipGeneratorMixin
+from indico.util.fs import secure_filename
+from indico.util.i18n import ngettext, _
+from indico.web.util import jsonify_data, jsonify_form
+
+
+class PaperJudgmentMixin:
+    """Perform bulk change state operations on selected papers"""
+
+    def _process(self):
+        form = BulkPaperJudgmentForm(event=self.event_new, judgment=request.form.get('judgment'),
+                                     contribution_id=[c.id for c in self.contributions])
+        if form.validate_on_submit():
+            judgment_data, contrib_data = form.split_data
+            submitted_papers = [c for c in self.contributions if
+                                c.paper_last_revision and c.paper_last_revision.state == PaperRevisionState.submitted]
+            for submitted_paper in submitted_papers:
+                judge_paper(submitted_paper, contrib_data, judge=session.user, **judgment_data)
+            num_submitted_papers = len(submitted_papers)
+            num_not_submitted_papers = len(self.contributions) - num_submitted_papers
+            if num_submitted_papers:
+                flash(ngettext("One paper has been judged.",
+                               "{num} paper have been judged.",
+                               num_submitted_papers).format(num=num_submitted_papers), 'success')
+            if num_not_submitted_papers:
+                flash(ngettext("One contribution has been skipped since it has no paper submitted yet or is in "
+                               "a final state.",
+                               "{num} contributions have been skipped since they have no paper submitted yet or are in "
+                               "a final state.",
+                               num_not_submitted_papers).format(num=num_not_submitted_papers), 'warning')
+            return jsonify_data(**self.list_generator.render_list())
+        return jsonify_form(form=form, submit=_('Judge'), disabled_until_change=False)
+
+
+class PapersDownloadAttachmentsMixin(ZipGeneratorMixin):
+    """Generate a ZIP file with paper files for a given list of contributions"""
+
+    def _prepare_folder_structure(self, item):
+        paper_title = secure_filename('{}_{}'.format(item.contribution.title, unicode(item.contribution.id)), 'paper')
+        file_name = secure_filename('{}_{}'.format(unicode(item.id), item.filename), item.filename)
```

Comment:
no need for secure_filename here (we already do it when saving), you fallback to a default containing the only user-provided part so it does almost nothing in the end